src/cell.cpp

```diff
@@ -334,12 +334,12 @@ double Cell::temperature(int32_t instance) const
 void Cell::set_temperature(double T, int32_t instance, bool set_contained)
 {
   if (settings::temperature_method == TemperatureMethod::INTERPOLATION) {
-    if (T < data::temperature_min) {
-      throw std::runtime_error {"Temperature is below minimum temperature at "
-                                "which data is available."};
-    } else if (T > data::temperature_max) {
-      throw std::runtime_error {"Temperature is above maximum temperature at "
-                                "which data is available."};
+    if (T < (data::temperature_min - settings::temperature_tolerance)) {
+      throw std::runtime_error {fmt::format("Temperature of {} K is below minimum temperature at "
+                                "which data is available of {} K.", T, data::temperature_min)};
+    } else if (T > (data::temperature_max + settings::temperature_tolerance)) {
+      throw std::runtime_error {fmt::format("Temperature of {} K is above maximum temperature at "
+                                "which data is available of {} K.", T, data::temperature_max)};
```

Comment:
I like this change. I'm wondering, is `data::temperature_min` the temperature you want to tell them about (which seems okay since it relates to the file we have), or is `data::tempterature_min - settings::temperature_tolerance` technically better since that is the lowest temp that the code will run for. Same for `data::temperature_max` vs `data::temperature_max + settings::temperature_tolerance`.

Reply:
I like showing the actual bounds of the data. I see the tolerancing as being a subtle way to prevent unnecessary crashes, rather than an intentional use. Also, I'd rather try to steer users into using real data, rather than data off the map.

Also, it's worth noting the min and max temperatures are global min and max, not necessarily the min and max available for the nuclides within the current cell being assigned a temperature. Technically, this could tell you a certain temperature is okay, even though the cell's nuclides don't actually have data at that temperature (we'd probably hit the same segfault described in #1292). This would be a pretty fringe case, so I think the single global min and max is fine.